woodwork/tests/test_utils.py

```diff
@@ -566,3 +568,315 @@ def test_parse_logical_type_errors():
+    assert to_pandas(combined_df).equals(pandas_combined_df)
+
+
+def test_concat_cols_with_series(sample_df):
+    df = sample_df[['id', 'full_name', 'email', 'phone_number', 'age']]
+    s1 = sample_df['signup_date']
+    sample_df.ww.init()
+    s2 = sample_df['is_registered']
+
+    combined_df = concat_columns([df, s1, s2])
+    assert combined_df.ww == sample_df.ww
+
+    df.ww.init()
+    s1.ww.init()
+    s2.ww.init()
+    pandas_combined_df = pd.concat([to_pandas(df), to_pandas(s1), to_pandas(s2)], axis=1, join='outer', ignore_index=False)
+    assert to_pandas(combined_df).equals(pandas_combined_df)
+
+
+def test_concat_cols_with_conflicting_ww_indexes(sample_df):
+    df1 = sample_df[['id', 'phone_number', 'email']]
+    df1.ww.init(index='id')
+    df2 = sample_df[['full_name', 'age', 'signup_date', 'is_registered']]
+    df2.ww.init(index='full_name')
+
+    error = ('Cannot have index columns set on more than one table in objs. '
+             'Please remove the index columns from all but one table.')
+    with pytest.raises(IndexError, match=error):
+        concat_columns([df1, df2])
+
+    df1 = sample_df[['id', 'phone_number', 'email']]
+    df1.ww.init(time_index='id')
+    df2 = sample_df[['full_name', 'age', 'signup_date', 'is_registered']]
+    df2.ww.init(time_index='signup_date')
+
+    error = ('Cannot have time_index columns set on more than one table in objs. '
+             'Please remove the time_index columns from all but one table.')
+    with pytest.raises(IndexError, match=error):
+        concat_columns([df1, df2])
+
+
+def test_concat_cols_with_ww_indexes(sample_df):
+    df1 = sample_df[['id', 'phone_number', 'email']]
+    df1.ww.init(index='id')
+    df2 = sample_df[['full_name', 'age', 'signup_date', 'is_registered']]
+    df2.ww.init(time_index='signup_date')
+
+    combined_df = concat_columns([df1, df2])
+    assert combined_df.ww.index == 'id'
+    assert combined_df.ww.time_index == 'signup_date'
+
+
+def test_concat_cols_with_duplicate_ww_indexes(sample_df):
+    df1 = sample_df[['id', 'phone_number', 'signup_date', 'email']]
+    df1.ww.init(index='id', time_index='signup_date')
+    df2 = sample_df[['full_name', 'age', 'id', 'signup_date', 'is_registered']]
+    df2.ww.init(index='id', time_index='signup_date')
+
+    error = ('Cannot have index columns set on more than one table in objs. '
+             'Please remove the index columns from all but one table.')
+    with pytest.raises(IndexError, match=error):
+        concat_columns([df1, df2])
+
+    new_df2 = df2.ww.drop(['id', 'signup_date'])
+
+    # Because underlying index is set, this won't change concat operation
+    pd.testing.assert_index_equal(to_pandas(df1.index), to_pandas(new_df2.index))
+
+    combined_df = concat_columns([df1, new_df2])
+    assert combined_df.ww.index == 'id'
+    assert combined_df.ww.time_index == 'signup_date'
+    assert len(set(combined_df.columns)) == len(set(sample_df.columns))
```

Comment:
Is the conversion to `set` here needed for anything?

Reply:
when we were allowing multiple index columns across dataframes, I the idea was to confirm that we had ended up without any of the duplicate columns (though this would have been caught by woodwork init), but since that behavior has changed, it's probably even less necessary